python/google/protobuf/descriptor.py

```diff
@@ -832,8 +832,11 @@ def __init__(self, name, package, options=None, serialized_pb=None,
     super(FileDescriptor, self).__init__(options, 'FileOptions')
 
     if pool is None:
-      from google.protobuf import descriptor_pool
-      pool = descriptor_pool.Default()
+      if api_implementation.Type() == 'cpp':
+        pool = _message.default_pool
```

Comment:
This code was here previously, so I was waiting on some confirmation that was the case. See previous line 850. I can remove if that's the consensus here.